distributed/distributed-schema.yaml

```diff
@@ -754,8 +754,14 @@ properties:
             properties:
               connect:
                 type: string
+                description: |
+                  Timeout after which to error when estabilishing a connection.
+                  For example, when creating a connection between client and worker,
+                  client and scheduler, etc.
               tcp:
                 type: string
+                description: |
+                  Timeout after which to error when creating a TCP/Socket connection
```

Comment:
This is not accurate. This timeout sets a couple of [kernel level timeouts](https://github.com/dask/distributed/blob/0ecc373260347b4410286714aed667fb51b61d51/distributed/comm/tcp.py#L65-L121) that take effect once a connection is established.

Specifically, it sets `TCP_USER_TIMEOUT` (See https://man7.org/linux/man-pages/man7/tcp.7.html) and configures a `KEEPALIVE` probe with appropriate intervals. I'm not sure if it is worth it to go into that much detail, though.

The combination of these two settings ensures that a TCP connection is automatically closed if the remote is dead, or rather, the kernel hasn't acknowledged any TCP package in `TCP_USER_TIMEOUT`s which very likely means the remote is dead.

We use this mechanism, for instance, to infer whether or not a worker died.